sqlglot/dialects/singlestore.py

```diff
@@ -64,6 +103,26 @@ class Parser(MySQL.Parser):
                 ),
                 format=MySQL.format_time(seq_get(args, 1)),
             ),
+            "BSON_EXTRACT_BSON": build_json_extract_path(exp.JSONBExtract),
+            "BSON_EXTRACT_STRING": build_json_extract_path(
+                exp.JSONBExtractScalar, json_type="STRING"
+            ),
+            "BSON_EXTRACT_DOUBLE": build_json_extract_path(
+                exp.JSONBExtractScalar, json_type="DOUBLE"
+            ),
+            "BSON_EXTRACT_BIGINT": build_json_extract_path(
+                exp.JSONBExtractScalar, json_type="BIGINT"
+            ),
+            "JSON_EXTRACT_JSON": build_json_extract_path(exp.JSONExtract),
+            "JSON_EXTRACT_STRING": build_json_extract_path(
+                exp.JSONExtractScalar, json_type="STRING"
+            ),
+            "JSON_EXTRACT_DOUBLE": build_json_extract_path(
+                exp.JSONExtractScalar, json_type="DOUBLE"
+            ),
+            "JSON_EXTRACT_BIGINT": build_json_extract_path(
+                exp.JSONExtractScalar, json_type="BIGINT"
+            ),
```

Comment:
Is there a variant of `JSON_EXTRACT` that works on all types? I'm wondering if we can produce a `CAST` to the `<type>` in `JSON_EXTRACT_<type>`, instead of storing that info in the `JSONExtract*` nodes.

Reply:
JSON_EXTRACT_JSON can extract any type, but casting of JSON object into a concrete datatype returns different results.
```
MySQL [(none)]> SELECT JSON_EXTRACT_JSON('{"a": "b"}', 'a') :> TEXT AS data;
+------+
| data |
+------+
| "b"  |
+------+
MySQL [(none)]> SELECT JSON_EXTRACT_STRING('{"a": "b"}', 'a') AS data;
+------+
| data |
+------+
| b    |
+------+
1 row in set (0.021 sec)
```

```
MySQL [(none)]> SELECT JSON_EXTRACT_JSON('{"a": null}', 'a') :> TEXT AS data;
+------+
| data |
+------+
| null |
+------+
1 row in set (0.001 sec)

MySQL [(none)]> SELECT JSON_EXTRACT_STRING('{"a": null}', 'a') AS data;
+------+
| data |
+------+
| NULL |
+------+
1 row in set (0.001 sec)

```